ourprojects/models.py

```diff
@@ -38,12 +78,100 @@ class Meta(Record.Meta, TracksRun.Meta, TracksUpdates.Meta):
+        ...     doi="10.1000/xyz123",
+        ...     preprint=False,
+        ...     journal="Nature Biotechnology",
+        ...     description="A groundbreaking research paper.",
+        ...     text="A really informative abstract.",
+        ...     published_at=date(2023, 11, 21),
+        ... ).save()
+    """
+
+    class Meta(Record.Meta, TracksRun.Meta, TracksUpdates.Meta):
+        abstract = False
+
+    id: int = models.AutoField(primary_key=True)
+    """Internal id, valid only in one DB instance."""
+    uid: str = CharField(unique=True, max_length=12, default=ids.base62_12)
+    """Universal id, valid across DB instances."""
+    """Universal id, valid across DB instances."""
+    name: str = CharField(db_index=True)
+    """Title or name of the reference document."""
+    abbr: str | None = CharField(
+        max_length=32,
+        db_index=True,
+        unique=True,
+        null=True,
+    )
+    """A unique abbreviation for the reference."""
+    url: str | None = URLField(null=True, blank=True)
+    """URL linking to the reference."""
+    pubmed_id: int | None = BigIntegerField(null=True)
+    """A PudMmed ID."""
+    doi: int | None = CharField(
+        null=True,
+        db_index=True,
+        validators=[
+            RegexValidator(
+                regex=r"^(?:https?://(?:dx\.)?doi\.org/|doi:|DOI:)?10\.\d+/.*$",
+                message="Must be a DOI (e.g., 10.1000/xyz123 or https://doi.org/10.1000/xyz123)",
+            )
+        ],
+    )
+    """Digital Object Identifier (DOI) for the reference."""
+    preprint: bool = BooleanField(default=False)
+    """Whether the reference is from a preprint."""
+    journal: str | None = TextField(null=True)
+    """Name of the journal."""
+    description: str | None = TextField(null=True)
+    """Description of the reference."""
+    text: str | None = TextField(null=True)
+    """Abstract or full text of the reference."""
+    published_at: date | None = DateField(null=True, default=None)
+    """Publication date."""
+    persons: Person = models.ManyToManyField(Person, related_name="reference_persons")
+    artifacts: Artifact = models.ManyToManyField(
+        Artifact, through="ArtifactReference", related_name="references"
+    )
+    """Artifacts labeled with this reference."""
+
+
+class ArtifactReference(Record, LinkORM, TracksRun):
+    id: int = models.BigAutoField(primary_key=True)
+    artifact: Artifact = ForeignKey(Artifact, CASCADE, related_name="links_reference")
+    reference: Reference = ForeignKey(Reference, PROTECT, related_name="links_artifact")
+    feature: Feature = ForeignKey(
+        Feature,
+        PROTECT,
+        null=True,
+        default=None,
+        related_name="links_artifactreference",
+    )
+    label_ref_is_name: bool | None = BooleanField(null=True, default=None)
+    feature_ref_is_name: bool | None = BooleanField(null=True, default=None)
+
+
 class ArtifactProject(Record, LinkORM, TracksRun):
```

Comment:
Remove this link model. If there's no additional fields (meaning contains only the foreign keys of the two linked models), django takes care of it automatically.